src/Ng/Jobs/LoopingNgJob.cs

```diff
@@ -38,8 +38,15 @@ protected LoopingNgJob(ILoggerFactory loggerFactory)
                 }
 
                 Logger.LogInformation("Running job.");
-                await RunInternal(cancellationToken);
-                Logger.LogInformation("Job finished!");
+                try
+                {
+                    await RunInternal(cancellationToken);
+                    Logger.LogInformation("Job finished!");
+                }
+                catch (TransientException te)
+                {
+                    Logger.LogWarning("Job execution ended due to a transient exception: {0}. {1}", te.Message, te.InnerException?.StackTrace??te.StackTrace);
```

Comment:
Well, previously, when the jobs encountered these issues, the process would crash and they would restart immediately. I think it's fine that we wait, however, because with no wait there's the potential for an infinite restart loop.